db/merge_helper.cc

```diff
@@ -268,7 +269,7 @@ Status MergeHelper::MergeUntil(InternalIterator* iter,
         if (keys_.size() == 1) {
           // we need to re-anchor the orig_ikey because it was anchored by
           // original_key before
-          Status pikStatus = ParseInternalKey(keys_.back(), &orig_ikey);
+          pikStatus = ParseInternalKey(keys_.back(), &orig_ikey);
           pikStatus.PermitUncheckedError();
```

Comment:
This check is unnecessary.  The assert line immediately after will take care of setting the checked flag

Reply:
We need coverage in non-debug build too - right?

Reply:
Yes. In fact I hope we can one day run the crash test on an optimized build of `db_stress` with `Status` check enforcement enabled. Despite I named the feature "`Status` check assertion", it actually no longer uses `assert()` to enfoce `Status`es are checked so does not require an assertion-enabled build.